bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/aws/http/AwsSalusApi.java

```diff
@@ -0,0 +1,184 @@
+import org.openhab.binding.salus.internal.rest.Device;
+import org.openhab.binding.salus.internal.rest.DeviceProperty;
+import org.openhab.binding.salus.internal.rest.GsonMapper;
+import org.openhab.binding.salus.internal.rest.RestClient;
+import org.openhab.binding.salus.internal.rest.exceptions.AuthSalusApiException;
+import org.openhab.binding.salus.internal.rest.exceptions.SalusApiException;
+import org.openhab.binding.salus.internal.rest.exceptions.UnsuportedSalusApiException;
+
+import software.amazon.awssdk.crt.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.crt.auth.signing.AwsSigner;
+import software.amazon.awssdk.crt.auth.signing.AwsSigningConfig;
+import software.amazon.awssdk.crt.auth.signing.AwsSigningResult;
+import software.amazon.awssdk.crt.http.HttpHeader;
+import software.amazon.awssdk.crt.http.HttpRequest;
+import software.amazon.awssdk.regions.Region;
+
+/**
+ * The SalusApi class is responsible for interacting with a REST API to perform various operations related to the Salus
+ * system. It handles authentication, token management, and provides methods to retrieve and manipulate device
+ * information and properties.
+ *
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public class AwsSalusApi extends AbstractSalusApi<Authentication> {
+    private final AuthenticationHelper authenticationHelper;
+    private final String companyCode;
+    private final String awsService;
+    private final Region region;
+    @Nullable
+    CogitoCredentials cogitoCredentials;
+
+    private AwsSalusApi(String username, byte[] password, String baseUrl, RestClient restClient, GsonMapper mapper,
+            Clock clock, AuthenticationHelper authenticationHelper, String companyCode, String awsService,
+            Region region) {
+        super(username, password, baseUrl, restClient, mapper, clock);
+        this.authenticationHelper = authenticationHelper;
+        this.companyCode = companyCode;
+        this.awsService = awsService;
+        this.region = region;
+    }
+
+    public AwsSalusApi(String username, byte[] password, String baseUrl, RestClient restClient, GsonMapper gsonMapper,
+            String userPoolId, String clientId, Region region, String companyCode, String awsService) {
+        this(username, password, baseUrl, restClient, gsonMapper, Clock.systemDefaultZone(),
+                new AuthenticationHelper(userPoolId, clientId, region.id()), companyCode, awsService, region);
+    }
+
+    @Override
+    protected void login() throws SalusApiException {
+        logger.debug("Login with username '{}'", username);
+        var result = authenticationHelper.performSRPAuthentication(username, password);
+        authentication = new Authentication(result.accessToken(), result.expiresIn(), result.tokenType(),
+                result.refreshToken(), result.idToken());
+        var local = LocalDateTime.now(clock).plusSeconds(authentication.expiresIn())
```

Comment:
Fix null warning by creating a local var
```suggestion
        AuthT authentication = new Authentication(result.accessToken(), result.expiresIn(), result.tokenType(),
                result.refreshToken(), result.idToken());
        var local = LocalDateTime.now(clock).plusSeconds(authentication.expiresIn())
        this.authentication = authentication;
```

Reply:
done